examples/step-1/step-1.cc

```diff
@@ -143,46 +143,58 @@ void second_grid ()
+      // triangulation aren't stored as a simple array, but the concept of an
+      // *iterator* generalizes how pointers work to arbitrary collections of
+      // objects (see <a href=
+      // "http://en.wikipedia.org/wiki/Iterator#C.2B.2B">wikipedia</a> for more
+      // information). Typically, any container type in C++ will return an
+      // iterator pointing to the start of the collection with a method called
+      // `begin`, and an iterator point to 1 past the end of the collection with
+      // a method called `end`. We can increment an iterator `it` with the
+      // operator `++it`, dereference it to get the underlying data with `*it`,
+      // and check to see if we're done by comparing `it != collection.end()`.
       //
       // The second important piece is that we only need the active cells.
       // Active cells are those that are not further refined, and the only
-      // ones that can be marked for further refinement, obviously. deal.II
-      // provides iterator categories that allow us to iterate over <i>all</i>
-      // cells (including the parent cells of active ones) or only over the
-      // active cells. Because we want the latter, we need to choose
-      // Triangulation::active_cell_iterator as data type.
+      // ones that can be marked for further refinement. deal.II provides
+      // iterator categories that allow us to iterate over <i>all</i> cells
+      // (including the parent cells of active ones) or only over the active
+      // cells. Because we want the latter, we need to call the method
+      // Triangulation<dim>::active_cell_iterators().
       //
-      // Finally, by convention, we almost always use the names
-      // <code>cell</code> and <code>endc</code> for the iterator pointing to
-      // the present cell and to the "one-past-the-end" iterator. This is, in
-      // a sense a misnomer, because the object is not really a "cell": it is
-      // an iterator/pointer to a cell. We should really have started to call
-      // these objects <code>cell_iterator</code> when deal.II started in
-      // 1998, but it is what it is.
+      // Putting all of this together, we can loop over all the active cells of
+      // a triangulation with
+      // @code{.cpp}
+      //     for (auto it = triangulation.active_cell_iterators().begin();
+      //          it != triangulation.active_cell_iterators().end();
+      //          ++it)
+      //       {
+      //         auto cell = *it;
+      //         // Then a miracle occurs...
+      //       }
+      // @endcode
+      // In the initializer of this loop, we've used the `auto` keyword for the
+      // type of the iterator `it`. The `auto` keyword means that the type of
+      // the object being declared will be inferred from the context. This
+      // keyword is useful when the actual type names are long or possibly even
+      // redundant. If you're unsure of what the type is and want to look up
```

Comment:
Can you add a sentence that the type here would be ``Triangulation<2>::active_cell_iterator``?